### PR TITLE
Use the baud rate given on the cmdline for getty

### PR DESCRIPTION
**Pull request by jbro**
Don't just default to 9600 Baud on serial consoles when given one with options on the Linux kernel command line.

Should work with the format given onhttps://www.kernel.org/doc/html/latest/admin-guide/serial-console.html

### CODE REVIEW

File: overlay/libexec/k3os/boot [hunk 1]
```diff
@@ -16,7 +16,13 @@ setup_ttys()
             ;;
         console=*)
             CONSOLE_SPEC=${x#console=}
-            IFS=, read TTY BAUDRATE _ <<<"${CONSOLE_SPEC},9600"
+            IFS=, read TTY BAUDRATE _ <<<"${CONSOLE_SPEC}"
+
+            if [ -n "$BAUDRATE" ]; then
+                BAUDRATE=$(grep -o '^[0-9]*' <<< "$BAUDRATE")
```
Comment (rgl) [inline overlay/libexec/k3os/boot:22]: Is this restricting BAUDRATE to be only numbers? If so, I think this should support stuff like `9600n8` (like described in the kernel page you've linked).

Comment (jbro) [reply overlay/libexec/k3os/boot:22]: The reason for only matching the numbers is that getty only expects numbers as the parameter where `$BAUDRATE` is used. As far as I can tell from the manual, getty is able to infer stop bits, checksum etc.

Comment (rgl) [reply overlay/libexec/k3os/boot:22]: AFICT the serial port has no auto-negotiation feature, so I'm now wondering how it does that magic!

Comment (jbro) [reply overlay/libexec/k3os/boot:22]: Well, getty only has a flag for "assume 8bit clean", it doesn't have a flag for 7bit with odd, even, none or space parity, so maybe it guesses and things just work. Anyway, the patch still makes things better, even if it assumes 8bit. In my case I was using a machine with coreboot hard-coded to 115200 baud, meaning I had to start my term i with the baud rate, and switch when the kernel started to see a complete boot log.

Comment (rgl) [reply overlay/libexec/k3os/boot:22]: Indeed it does, no doubt about that :-)

Coreboot? Woah! What machine do you have? I've very curious who is using coreboot :-)

Comment (jbro) [reply overlay/libexec/k3os/boot:22]: It's an https://pcengines.ch/apu2c4.htm, it's a very nice machine I can only recommend them :-)

Comment (rgl) [reply overlay/libexec/k3os/boot:22]: Nice! I've been looking at the apu4d4 model, but I still can't justify getting one eheh

**Pull request merged**